### PR TITLE
Improve Snippet Generator documentation

### PR DESCRIPTION
**Pull request by orevial**
This PR aims to improve the current documentation & snippet generation available from Jenkins. Basically, the input in the snippet generator will look like that :
marathon-plugin-snippetgenerator_inputs

And the associated output like that :
marathon-plugin-snippetgenerator_output

A few things to note :
1. I used existing `MarathonUri `and `MarathonLabel ` classes into `MarathonStep` instead of raw `List/Map<String>`, which is both more consistent with what was used in `MarathonRecorder`, and easier to document using the snippet generator. 

1. I added URL/URIs validation for both the `MarathonRecorder `and the `MarathonStep` (in documentation)

1. I renamed the field `appid` to `id` in the documentation to be more consistent with code (as `appid `is deprecated)

### CODE REVIEW

**Comment by orevial** [comment]
I'm not sure I understand Jenkins failed check output, but from what I get it's an error with Javadoc parameters, which should not be linked with my commit (also, the same error was thrown in the 2 previous visible builds)

**Comment by colin-msphere** [comment]
Thank you for the pull request. I am looking in to the jenkins-ci doc linting errors.

**Comment by colin-msphere** [comment]
ok to test

File: src/main/resources/com/mesosphere/velocity/marathon/MarathonStep/help-docker.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div>
+    <p>
+        A Docker image name [including a URI if using a private registry]. This will overwrite the <i>image</i> field in
```
Comment (colin-msphere) [inline src/main/resources/com/mesosphere/velocity/marathon/MarathonStep/help-docker.html]: Please remove the `[]`s:

> A Docker image name, including a URI if using a private registry.

**Comment by OlivierRevial** [comment]
Just rebased, and removed help file braces as per your recommandation.

**Comment by colin-msphere** [comment]
retest this please

**Pull request merged**